hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/shell/TestBulkDeleteCommand.java

```diff
@@ -27,103 +27,134 @@
-        FSDataOutputStream fsDataOutputStream = lfs.create(fileWithDeletePaths, true);
-        BufferedWriter br = new BufferedWriter(new OutputStreamWriter(fsDataOutputStream));
-        for(int i = 0; i < 100; i++) {
-            Path p = new Path(baseDir, baseFileName + i);
-            lfs.create(p);
-            br.write(p.toUri().toString());
-            br.newLine();
-        }
-        br.flush(); // flush the file to write the contents
-        br.close(); // close the writer
-        List<String> finalCommandList = new ArrayList<>();
-        finalCommandList.add("-bulkDelete");
-        finalCommandList.add("-readFromFile");
-        finalCommandList.add(fileWithDeletePaths.toUri().toString());
-        finalCommandList.add(baseDir.toUri().toString());
-        shell.run(finalCommandList.toArray(new String[0]));
-        Assertions.assertThat(lfs.listFiles(baseDir, false).hasNext())
-                .as("All the files should have been deleted").isEqualTo(false);
+  }
 
+  @Test
+  public void testLocalFileDeletionWithFileName() throws IOException {
+    String deletionDir = "toDelete";
+    String baseFileName = "file_";
+    Path baseDir = new Path(testRootDir, deletionDir);
+    Path fileWithDeletePaths = new Path(testRootDir, "fileWithDeletePaths");
+    FSDataOutputStream fsDataOutputStream = lfs.create(fileWithDeletePaths, true);
+    BufferedWriter br = new BufferedWriter(new OutputStreamWriter(fsDataOutputStream));
+    for (int i = 0; i < 100; i++) {
+      Path p = new Path(baseDir, baseFileName + i);
+      ContractTestUtils.touch(lfs, p);
+      br.write(p.toUri().toString());
+      br.newLine();
     }
+    br.flush(); // flush the file to write the contents
+    br.close(); // close the writer
+    List<String> finalCommandList = new ArrayList<>();
+    finalCommandList.add("-bulkDelete");
+    finalCommandList.add("-readFromFile");
+    finalCommandList.add(fileWithDeletePaths.toUri().toString());
+    finalCommandList.add(baseDir.toUri().toString());
+    shell.run(finalCommandList.toArray(new String[0]));
+    Assertions.assertThat(lfs.listFiles(baseDir, false).hasNext())
+            .as("All the files should have been deleted under the path:" +
+                    baseDir).isEqualTo(false);
+
+  }
+
+  @Test
+  public void testWrongArgumentsWithNonChildFile() throws IOException, URISyntaxException {
+    BulkDeleteCommand bulkDeleteCommand = new BulkDeleteCommand(conf);
+    LinkedList<String> arguments = new LinkedList<>();
+    String arg1 = "file:///file/name/1";
+    String arg2 = "file:///file/name";
+    arguments.add(arg1);
+    arguments.add(arg2);
+    LinkedList<PathData> pathData = bulkDeleteCommand.expandArguments(arguments);
+    Assertions.assertThatThrownBy(() -> bulkDeleteCommand.processArguments(pathData)).
+                                  describedAs("Child paths must be contained inside the base path").
+                                  isInstanceOf(IOException.class);
```

Comment:
match the message what must be thrown in case of different argument errors? Just IOE is vague.

Reply:
...but there's a risk that different implementations may raise different subclasses -I don't think we are that strict in the specification about what to return